File: wear/src/main/java/info/nightscout/androidaps/interaction/menus/StatusMenuActivity.java
package info.nightscout.androidaps.interaction.menus;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import info.nightscout.androidaps.R;
import info.nightscout.androidaps.data.ListenerService;
import info.nightscout.androidaps.interaction.utils.MenuListActivity;

/**
 * Created by adrian on 09/02/17.
 */

public class StatusMenuActivity extends MenuListActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTitle(R.string.menu_status);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected List<MenuItem> getElements() {
        List<MenuItem> menuitems = new ArrayList<>();
        menuitems.add(new MenuItem(R.drawable.ic_status, getString(R.string.status_pump)));
        menuitems.add(new MenuItem(R.drawable.ic_loop_closed, getString(R.string.status_loop)));
        menuitems.add(new MenuItem(R.drawable.ic_status, getString(R.string.status_cpp)));
        menuitems.add(new MenuItem(R.drawable.ic_tdd, getString(R.string.status_tdd)));

        return menuitems;
    }

    @Override
    protected void doAction(String action) {
        if (getString(R.string.status_pump).equals(action)) {
            ListenerService.initiateAction(this, "status pump");
        } else if (getString(R.string.status_loop).equals(action)) {
            ListenerService.initiateAction(this, "status loop");
        } else if (getString(R.string.status_cpp).equals(action)) {
            ListenerService.initiateAction(this, "opencpp");
        } else if (getString(R.string.status_tdd).equals(action)) {
            ListenerService.initiateAction(this, "tddstats");
        }
    }
}

File: wear/src/main/java/info/nightscout/androidaps/interaction/menus/StatusMenuActivity.kt
package info.nightscout.androidaps.interaction.menus

import android.os.Bundle
import app.aaps.core.interfaces.rx.events.EventWearToMobile
import app.aaps.core.interfaces.rx.weardata.EventData.ActionLoopStatus
import app.aaps.core.interfaces.rx.weardata.EventData.ActionPumpStatus
import app.aaps.core.interfaces.rx.weardata.EventData.ActionTddStatus
import info.nightscout.androidaps.R
import info.nightscout.androidaps.interaction.utils.MenuListActivity

class StatusMenuActivity : MenuListActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        setTitle(R.string.menu_status)
        super.onCreate(savedInstanceState)
    }

    override fun provideElements(): List<MenuItem> =
        ArrayList<MenuItem>().apply {
            add(MenuItem(R.drawable.ic_status, getString(R.string.status_pump)))
            add(MenuItem(R.drawable.ic_loop_closed, getString(R.string.status_loop)))
            add(MenuItem(R.drawable.ic_tdd, getString(R.string.status_tdd)))
        }

    override fun doAction(position: String) {
        when (position) {
            getString(R.string.status_pump) -> rxBus.send(EventWearToMobile(ActionPumpStatus(System.currentTimeMillis())))
            getString(R.string.status_loop) -> rxBus.send(EventWearToMobile(ActionLoopStatus(System.currentTimeMillis())))
            getString(R.string.status_tdd)  -> rxBus.send(EventWearToMobile(ActionTddStatus(System.currentTimeMillis())))
        }
    }
}
